third_party/rust/challenge_bypass_ristretto_cxx/src/lib.rs

```diff
@@ -0,0 +1,598 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use core::str::Utf8Error;
+
+use challenge_bypass_ristretto::{errors, voprf};
+use cxx::{CxxString, CxxVector};
+use derive_more::From;
+use hmac::Hmac;
+use lazy_static::lazy_static;
+use rand::rngs::OsRng;
+use sha2::Sha512;
+
+type HmacSha512 = Hmac<Sha512>;
+
+lazy_static! {
+    static ref NO_ERROR: Error = Error::none();
+}
+
+macro_rules! impl_result {
+    ($t:ident, $r:ident) => {
+        impl $r {
+            fn error(self: &$r) -> &Error {
+                match &self.0 {
+                    Err(e) => &e,
+                    Ok(_) => &NO_ERROR,
+                }
+            }
+
+            fn is_ok(self: &$r) -> bool {
+                self.error().is_ok()
+            }
+
+            fn unwrap(self: &$r) -> &$t {
+                self.0.as_ref().expect(
+                    "Unhandled error before unwrap call, self->error().code != TokenError::None",
+                )
+            }
+        }
+    };
+}
+
+macro_rules! impl_decode_base64 {
+    ($f:ident, $t:ident, $r:ident) => {
+        fn $f(s: &str) -> Box<$r> {
+            Box::new(|| -> Result<$t, Error> { Ok(voprf::$t::decode_base64(&s)?.into()) }().into())
+        }
+    };
+}
+
+#[allow(unsafe_op_in_unsafe_fn)]
+#[cxx::bridge(namespace = cbr_cxx)]
+pub mod ffi {
```

Comment:
This would differ from how other Rust ffi libraries do it. For example brave wallet has these:

```
#[allow(unsafe_op_in_unsafe_fn)]
#[cxx::bridge(namespace =  brave_wallet)]
mod ffi {
```

Reply:
Feels like cargo cult (ahem) to me, but it's not a big deal.